src/main/java/org/radarcns/management/domain/SensorData.java

```diff
@@ -204,32 +204,61 @@ public void setDeviceTypes(Set<DeviceType> deviceTypes) {
 
     @Override
     public boolean equals(Object o) {
```

Comment:
I usually use a template of
```java
if (this == o) {
        return true;
}
if (o == null || getClass() != o.getClass()) {
    return false;
}

SensorData that = (SensorData) o;
return Objects.equals(a, that.a)
    && ...;
```
using `getClass()` instead of `instanceof` makes the equals operation _symmetric_ (`a = b` <=> `b = a`)
Using `Objects.equals` eliminates a lot of `null` checking.

Similarly, `Objects.hash(propA, propB, ...)` takes a varargs of all properties, making the `hashCode` just a list of properties. If `id` is intended to be non-null and unique, you could just return `id` in hashCode. Note that `a = b` => `a.hash() = b.hash()` but not the reverse, so hashCode may have more clashes, as long as the clashes remain infrequent.